packages/headless/src/controllers/facets/automatic-facet-builder/headless-automatic-facet-builder.test.ts

```diff
@@ -0,0 +1,38 @@
+import {configuration} from '../../../app/common-reducers';
+import {setDesiredCount} from '../../../features/facets/automatic-facet-set/automatic-facet-set-actions';
+import {automaticFacetSetReducer as automaticFacetSet} from '../../../features/facets/automatic-facet-set/automatic-facet-set-slice';
+import {searchReducer as search} from '../../../features/search/search-slice';
+import {MockSearchEngine, buildMockSearchAppEngine} from '../../../test';
+import {
+  AutomaticFacetBuilderProps,
+  AutomaticFacetBuilder,
+  buildAutomaticFacetBuilder,
+} from './headless-automatic-facet-builder';
+
+describe('automatic facets', () => {
+  let engine: MockSearchEngine;
+  let automaticFacets: AutomaticFacetBuilder;
+  let props: AutomaticFacetBuilderProps;
+
+  beforeEach(() => {
+    props = {desiredCount: 5};
+    engine = buildMockSearchAppEngine();
+    automaticFacets = buildAutomaticFacetBuilder(engine, props);
+  });
+
+  it('should add the correct reducers to engine', () => {
+    expect(engine.addReducers).toHaveBeenCalledWith({
+      automaticFacetSet,
+      configuration,
+      search,
+    });
+  });
+
+  it('should dispatch #setDesiredCount', () => {
+    expect(engine.actions).toContainEqual(setDesiredCount(props.desiredCount));
+  });
+
+  it('should return automatic facets as empty array if the response is empty', () => {
+    expect(automaticFacets.state.automaticFacets).toEqual([]);
```

Comment:
I believe this confirms my previous suspicion that `automaticFacets` _**should not**_ be nullable then, right? It changes very little, but if the expected outcome for an empty response is an empty array (rather than nothing at all), then there's no situation (that I can think of) where `automaticFacets` could be `null`/`undefined` in the state 👀 

If I'm overlooking something though, please let me know!

Reply:
Yea you are right, it is better if it is not nullable everywhere and a empty array is created after the request. I changed it